Add doorsense sensor for August 3rd Gen Smart Lock Pro

ehendrix23 opened the pull request:
## Description:
The August 3rd Gen Smart Lock Pro comes with a DoorSense sensor to identify if the door is open or closed.

I added a new binary sensor within the august component to expose the sensor to HASS. A binary sensor will be created for each lock that has the doorsense sensor available as well. During initialization it checks if the status returned is not Unknown and only if it is Unknown it will add the device, otherwise the binary sensor will not be added.

I further also added debug logging in various areas of the coding for both locks and doorbells.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#6488

## Checklist:
  - [X ] The code change is tested and works locally.
  - [X ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

StevenLooman committed:
Fix message "Updating dlna_dmr media_player took longer than ..."

StevenLooman committed:
Upgrade to async_upnp_client 0.12.4

StevenLooman committed:
Working on IGD

StevenLooman committed:
Merge remote-tracking branch 'upstream/dev' into igd

StevenLooman committed:
Working on IGD

StevenLooman committed:
Merge remote-tracking branch 'upstream/dev' into igd

StevenLooman committed:
Working on IGD

StevenLooman committed:
Merge remote-tracking branch 'upstream/dev' into igd

StevenLooman committed:
Ensure config_flow is imported

StevenLooman committed:
Prevent error message when component is actually ok

StevenLooman committed:
Fix not being able to re-add IGD when removed

StevenLooman committed:
Working on igd

StevenLooman committed:
Move ensure_domain_data

StevenLooman committed:
Linting

StevenLooman committed:
Changes after review

StevenLooman committed:
Working on igd

StevenLooman committed:
Set ports (hass -> hass) when no ports have been specified

StevenLooman committed:
Fix uppercase

Fix translation

StevenLooman committed:
Rename igd to upnp

StevenLooman committed:
Fix tracebacks

StevenLooman committed:
Fix discovery/config entry handlers

StevenLooman committed:
Update requirements_all.txt

StevenLooman committed:
Fixes after rename

StevenLooman committed:
Preserve compatibility with original upnp

StevenLooman committed:
Merge remote-tracking branch 'upstream/dev' into igd

StevenLooman committed:
Changes after review by @MartinHjelmare

StevenLooman committed:
Changes after review by @MartinHjelmare

StevenLooman committed:
Fix stale docstrings

ehendrix23 committed:
Add doorsense sensor for August 3rd Gen Smart Lock Pro

Added the doorsense sensor that is part of the August 3rd Gen Smart Lock Pro as a binary sensor.
A binary sensor will be created for each operable lock discovered where a status of Open or Closed is retrieved during initialization, to prevent the binary sensor from being created if the doorsense sensor is not there.

Added debug logging in various areas for both locks and doorbells coding for August.

ehendrix23 committed:
Add doorsense sensor for August 3rd Gen Smart Lock Pro

Added the doorsense sensor that is part of the August 3rd Gen Smart Lock Pro as a binary sensor.
A binary sensor will be created for each operable lock discovered where a status of Open or Closed is retrieved during initialization, to prevent the binary sensor from being created if the doorsense sensor is not there.

Added debug logging in various areas for both locks and doorbells coding for August.

homeassistant commented:
Hi @ehendrix23, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#17116).

Once you do that we will be able to review and accept this pull request.

Thanks!

mvn23 committed:
Fix counter restore. (#17101)

Add config option to disable restore (always use initial value on restart).
Add unit tests for restore config option.

dgomes committed:
Merge pull request #16300 from StevenLooman/igd

Adds discovery and config flow
Breaking change: no longer possible to map ports other than the port used by Home Assistant

ehendrix23 committed:
Fixed else after return

Fixed the else statement after a return as per pylint.

ehendrix23 committed:
Fixed else after return

Fixed the else statement after a return as per pylint.

ehendrix23 committed:
Fixed invalid syntax

Fixed invalid syntax reported by houndci-bot and recheched locally with flake8, pylint, and pydocstyle.

ehendrix23 committed:
Fixed invalid syntax

Fixed invalid syntax reported by houndci-bot and recheched locally with flake8, pylint, and pydocstyle.

no2chem committed:
Support multiple accounts in Tile, use device identifiers (#17108)

* :sparkles: Update tile to support multiple accounts

* :art: fix indent

* :bug: fix format string

* :art: use .format

* :art: fix line indent

bachya committed:
Bumps simplisafe-python to 3.1.2 (#16931)

* Bumps simplisafe-python to 3.0.4

* Updated requirements

* Refresh token logic added

* Member-requested changes

* Removed unused import

* Updated CODEOWNERS

* Bump library to 3.1.2

scop committed:
Upgrade pytest to 3.8.2 (#17125)

MartinHjelmare committed:
Fix sonos async use (#17099)

* Entry setup wasn't using the async api. Fix this by using correct
  async api.s
* Also use new async executor scheduler in async_added_to_hass.

emlove committed:
Homekit controller reconnect (#17060)

* Add threaded call_later helper

* Reconnect to device when connection fails

* Consolidate connection logs and warn on first

Danielhiversen committed:
Tibber component and notify (#17062)

* Refactor tibber, and Tibber notify

* update Tibber lib.

* tibber

* Tibber coveragerc

* Tibber upgrade lib

* style

* comments

* use async_get_service

* event

jeradM committed:
Add time created to persistent notifications. (#17121)

* Add time created to persistent notifications.

* UTC

balloob committed:
Make it easier for auth to consume newer formats (#17127)

hthiery committed:
Add new component fritzbox binary_sensor (#17057)

* Add new component fritzbox binary_sensor

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>

* fix failed flake8 test

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>

* add new file to .coveragerc

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>

* use wildcard to cover all platforms

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>

* remove polling because polling is true by default

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>

* add blank line to keep imports ordered

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>

* Minor changes

Danielhiversen committed:
Update xiaomi lib (#17129)

arsaboo committed:
Add update service to Google Travel Sensor (#17092)

* Add update service

* Update service name

* Register service only once

* Make hound happy

awarecan committed:
Fix upnp component l10n error (#17132)

tsvi committed:
Take timezone into consideration when calulating Zmanim. Partial fix for #16946 (#17131)

balloob committed:
Allow config entry setup to raise not ready (#17135)

kirichkov committed:
Webhook component - pass headers to webhook handler (#17091)

* Pass headers to webhook handler

* Refactors webhook component to post Request object instead of data

* Update webhook tests

* Cleanup webhook test and fix a bug in ifttt

* Address code review comments

anapaulagomes committed:
Add timeout and fix oscillations on Samsung TV component (#17102)

* Add timeout and fix oscillations

* Adjust code to py3.5.3

* Clean code

balloob committed:
Remove discovery (#17070)

cdce8p committed:
Bugfix invalid entity_config parameter HomeKit (#17143)

persandstrom committed:
verisure configurable polling (#17144)

flowolf committed:
YesssSMS handling more errors, upgrade to version 0.2.3 (#17052)

* YesssSMS handling more errors, upgrade to version 0.2.1

 - handling missing internet connection nicely
 - disabling login with non-working credentials (website locked account for 1 hour)
 - upgrade to new upstream version of YesssSMS

* notify.yessssms tests

* test requirements

* flake8 fix

* fixing tests, new upstream version 0.2.3

fixing tests based on requested changes, coverage

* removing unmotivated print

* passing exception to ConnectionError and SMSSendingError logger

bachya committed:
Added OpenUV CODEOWNERS info (#17149)

gerard33 committed:
Fix a typo (#17147)

lwis committed:
Cleanly stop tradfri on shutdown (#17114)

* Tradfri shutdown fix

* Bump version

* Bump version

* Fix

* Derp

* Remove unnecessary shutdown event

rohankapoorcom committed:
Fix zoneminder zms_url construction (#17150)

Julius2342 committed:
Added warning to HomeKit component (#16807)

* added warning if more then 100 devices are added to HomeKit

cdce8p committed:
Add faucet, shower, sprinkler, valve to HomeKit (#17145)

PaulAnnekov committed:
Fix miflora connection errors during platform setup (#16798)

* possible fix for startup delay

* fixed reported issues

* moved update code into setup

* reverted to previous solution

prophit987 committed:
Discover Danfoss/devolo RS Room Sensor thermostat (#17153)

Allows the Danfoss/Devolo RS Room Sensor thermostat to be discovered as climate device in homeassistant.
This device is defined as GENERIC_TYPE_MULTILEVEL_SENSOR, and not GENERIC_TYPE_THERMOSTAT.
Ref: https://community.home-assistant.io/t/devolo-zwave-thermostat/58739

prophit987 committed:
@danielhiversen as codeowner for Tibber (#17154)

balloob committed:
Update frontend to 20181005.0

balloob committed:
Update translations

balloob committed:
Bumped version to 0.81.0.dev0

balloob committed:
Fix incorrect yaml in hangouts (#17169)

kennedyshead committed:
Adding myself as melissa owner (#17157)

fabaff committed:
Update core, add myself and introduce grouping

fabaff committed:
Revert "Update core, add myself and introduce grouping"

This reverts commit 13106a9b55c0dddb87d5a482ca9a1904911158b1.

balloob committed:
Fix data used for logbook (#17172)

* Fix data used for logbook

* Lint

cdce8p committed:
Fix device_tracker service call & cleanup (#17173)

* Bugfix group service - device_tracker

* Cleanup

fabaff committed:
Update core, add myself and introduce grouping (#17175)

fabaff committed:
Add myself to more sensors (#17185)

amelchio committed:
Upgrade aiolifx_effects to 0.2.1 (#17188)

persandstrom committed:
Verisure standard config for scan interval (#17192)

* verisure configurable polling

* fix indentation

GuyKh committed:
Upgrade boto3 to 1.9.16 (#17140)

cdce8p committed:
Bugfix switch flux - light service call (#17187)

* Bugfix switch flux - light service call

* Change x_val and y_val test

ehendrix23 committed:
Fix AttributeError NoneType

Upon startup of HASS there would be a traceback for the lock stating that AttributeError NoneType has no attribute 'battery_level'.
This change checks to ensure that self._lock_detail is not none; if it is then None is returned for atribute battery level otherwise battery level is returned.

ehendrix23 committed:
Fix AttributeError NoneType

Upon startup of HASS there would be a traceback for the lock stating that AttributeError NoneType has no attribute 'battery_level'.
This change checks to ensure that self._lock_detail is not none; if it is then None is returned for atribute battery level otherwise battery level is returned.

PhilRW committed:
Fix Todoist custom project update (#17115)

* Fix Todoist custom project update

Custom projects were not refreshing the API state and were using
local/stored state. This resulted in invalid tasks being retained upon
update. This change resets the local Todoist API state, syncs it, and
then continues normal update operation(s) on the Todoist project data
object.

* Remove blank line after docstring

* Update logging call

* Simplify logging

rohankapoorcom commented on homeassistant/components/august.py:
```diff
@@ -218,17 +223,25 @@ def get_doorbell_detail(self, doorbell_id):
         return self._lock_status_by_id.get(lock_id)
 
+    def get_lock_door_status(self, lock_id):
+        """Return door lock status."""
```
Personally, I would make this comment (and the one on line 236 a little more descriptive, it's unclear to me what the difference is between `lock status` and `door lock status`.

ehendrix23 replied:
Updated comments for both to better explain the difference. Please check if this explains it better (I know what I'm saying <g>).

persandstrom committed:
vsure version 1.5.0 (#17209)

amelchio committed:
Disallow list/dict for string configuration (#17202)

wonderslug committed:
Enable new registry rename for Insteon (#17171)

* Enable new registry rename for Insteon

* Segment unique_id from name

emontnemery committed:
Fix aliases support for RFLink sensors (#17190)

Danielhiversen committed:
Add forecast for Met.no (#17109)

test met

met no 0.3.0

fix line length

fix line length

Commit:
Add support for ElkM1 alarm/automation panel (#16952)

* Add support for ElkM1 alarm/automation panel.

* fix lines too long

* Address PR comments

* Fix hound ci errors

* Changes for PR comments

* Use vol.Range for checking range value

* Address PR comments

* Fix lint errors

* Added elkm1-lib requirement

* Update coverage to exclude elk

* Fix flake8 errors

* Fix flake8 error

* Cleanup config parsing

* Add housecode converter

* fix PR comments

* fix syntax error

* Fix PR comment

amelchio committed:
Add config entry for LIFX (#17201)

* Add config entry for LIFX

* Use list for dependencies

* Obsolete the platform config

* Use DOMAIN

* Use async_create_task

balloob committed:
Update frontend to 20181007.0

balloob committed:
Update translations

mbrrg committed:
Init sub-components using global var. (#17220)

Danielhiversen committed:
danielhiversen as codeowner for met.no (#17232)

balloob committed:
Prevent accidental device reg override (#17136)

syssi committed:
Add some new model names of Xiaomi Aqara devices (#17234)

* Add additional model name of the Xiaomi Aqara Button (WXKG11LM)

* Add additional model name of the Xiaomi Aqara Wireless Switch (WXKG02LM, WXKG03LM)

* Bump PyXiaomiGateway version

Commit:
Pushed to version 0.7.6 of denonavr library to add more sound modes (#17227)

balloob committed:
Fix SPC (#17236)

syssi committed:
Fix data_key and power_consumed attribute of the Aqara Wall Switch (Closes: #16457) (#17235)

Commit:
Water heater support (#17058)

* Moved econet to water_heater

* Wink and Econet are now in water heater!

* Removed away mode from econet and added demo water heater

* Added demo tests

* Updated coveragerc

* Fix lint issues.

* updated requirements all

* Requirements all actually updated.

* Reset wink and econet and fixed service.

* Reset wink and econet to the correct dev state

* Reset requirements_all and .coveragerc and removed the new econet and wink water_heater files

* Removed @bind_hass service methods

* Actually reset the .coverage file

* Fixed the tests

* Addressed @MartinHjelmare's comments

* Removed unused import

* Switched to async_add_executor_job

* Fixed lint

* Removed is_on

* Added celsius demo water heater and tests.

* Removed metric import

MatteGary committed:
Daikin Climate - Better integration with Climate base component (#16913)

* Daikin Climate - Better integration with Climate base component

Made some modification in order to better integrate the Daikin AC Component with the base Climate Component.
Benefits are:
Support localization for Operation Mode
Support for Homekit Integration (if the AC is turned On, now the status is updated in Homekit)

* Build bug fixing

* Bug fixing in Set Operation_Mode functionality

Fixed to .title() functionality in matching the Operation_Mode

* Fix useless code

* Fix hound bug

* Bug fixing

Change in list of Operation Mode

* Trailing white space fix

* Compile fixing

* Whitespace fix

OttoWinter committed:
Fix potential MQTT discovery race condition (#17208)

* Fix potential MQTT discovery race condition

* Rename data key

dlashua committed:
Add Support for Xiaomi Vibration Sensor (#16422)

OttoWinter committed:
Implement base for MQTT device registry integration (#16943)

* Implement base for MQTT device registry integration

* Lint

* Lint

* Address comments

* Lint

* Lint

* Address comments

* Only add keys if specified

See https://github.com/home-assistant/home-assistant/pull/17136#discussion_r223267185

balloob committed:
Guard for bad device info (#17238)

exxamalte committed:
updated georss-client library to 0.3 (#17239)

schmittx committed:
MyQ cover return unknown state if not available (#17207)

* Add additional supported states

* Use get method for lookup

* Return None if unable to get status

OttoWinter committed:
Add device registry to MQTT binary sensor (#17243)

OttoWinter committed:
Add device registry to MQTT switches (#17244)

OttoWinter committed:
Convert MQTT fan to config entry (#17247)

emontnemery committed:
Add support for HS color to mqtt light (#16958)

* Add support for HS color to mqtt light

* Warn if hs state update is invalid

ehendrix23 committed:
Updates based on review feedback

Removed unnecessary log entries.
Improved description for difference between get_lock_status and get_lock_door_status
Inverted order argument for _LOGGER in binary_sensor/august.py
Return nothing if _lock_detail is None otherwise return the battery level attribute now.
Getting lock status last now for lock, seemed to work better as otherwise sometimes would have issue with door sensor.

ehendrix23 committed:
Updates based on review feedback

Removed unnecessary log entries.
Improved description for difference between get_lock_status and get_lock_door_status
Inverted order argument for _LOGGER in binary_sensor/august.py
Return nothing if _lock_detail is None otherwise return the battery level attribute now.
Getting lock status last now for lock, seemed to work better as otherwise sometimes would have issue with door sensor.

ehendrix23 commented on homeassistant/components/august.py:
```diff
@@ -102,7 +102,7 @@ def setup_august(hass, config, api, authenticator):
         authentication = authenticator.authenticate()
     except RequestException as ex:
-        _LOGGER.error("Unable to connect to August service: %s", str(ex))
+        _LOGGER.error("Unable to connect to August service: %s.", str(ex))
```
Removed periods at the end of all the _LOGGER messages.

ehendrix23 commented on homeassistant/components/august.py:
```diff
@@ -156,6 +157,7 @@ class AugustData:
 
     def __init__(self, api, access_token):
         """Init August data object."""
+        _LOGGER.debug("Initializing data.")
```
Removed.

ehendrix23 commented on homeassistant/components/binary_sensor/august.py:
```diff
@@ -4,11 +4,15 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/sensor.august/
 """
+
```
Removed.

ehendrix23 commented on homeassistant/components/binary_sensor/august.py:
```diff
@@ -60,24 +69,53 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
+            if SENSOR_TYPES[sensor_type][1] == 'door':
+                continue
+            _LOGGER.debug("Adding doorbell sensor class %s for %s.",
+                          doorbell.device_name, SENSOR_TYPES[sensor_type][1])
```
Inverted order, thanks.

ehendrix23 commented on homeassistant/components/binary_sensor/august.py:
```diff
@@ -88,10 +126,18 @@ def device_class(self):
+            _LOGGER.debug("%s not changed; current state is %s.",
+                          self.entity_id, self._state)
+        else:
+            _LOGGER.debug("%s changed from %s to %s.",
```
Removed.

ehendrix23 commented on homeassistant/components/lock/august.py:
```diff
@@ -39,14 +43,17 @@ def __init__(self, data, lock):
 
     def lock(self, **kwargs):
         """Lock the device."""
+        _LOGGER.debug("Locking %s.", self.entity_id)
```
Removed.

ehendrix23 commented on homeassistant/components/lock/august.py:
```diff
@@ -39,14 +43,17 @@ def __init__(self, data, lock):
 
     def unlock(self, **kwargs):
         """Unlock the device."""
+        _LOGGER.debug("Unlocking %s.", self.entity_id)
```
Removed.

ehendrix23 commented on homeassistant/components/lock/august.py:
```diff
@@ -58,6 +65,13 @@ def update(self):
+            _LOGGER.debug("%s not changed; current status is %s.",
+                          self.entity_id, self._lock_status)
+        else:
+            _LOGGER.debug("%s changed from %s to %s.",
```
Removed.

ehendrix23 commented on homeassistant/components/lock/august.py:
```diff
@@ -77,6 +91,11 @@ def changed_by(self):
+        if self._lock_detail is not None:
+            battery_level = self._lock_detail.battery_level
+        else:
+            battery_level = None
```
Changed, will now not return anything if _lock_detail is None; otherwise will return the attribute.

ehendrix23 commented:
Thank you for all the comments in the review. I went through and believe to have addressed all of them. I also checked and after changes it is still working locally on my HASS. :-)

exxamalte committed:
emptying device state attributes if the update from the feed fails (#17249)

OttoWinter committed:
Add device registry to MQTT cover (#17245)

* Add device registry to MQTT cover

* Fix tests

balloob committed:
Add group foundation (#16935)

Add group foundation

Commit:
Add Elk light platform (#17222)

* Add Elk light platform.

* Add ElkM1 light code. Doh.

* Fix PR comments.

* Fix hound errors.

* Fix PR comment.

* Move state from base to class(s) where used

ehendrix23 committed:
Periods and return None

Added period in comment sections.
Missed one logger line that still had a period
Changed Return to Return None for lock attributes.

ehendrix23 committed:
Periods and return None

Added period in comment sections.
Missed one logger line that still had a period
Changed Return to Return None for lock attributes.

ehendrix23 commented on homeassistant/components/august.py:
```diff
@@ -218,17 +221,31 @@ def get_doorbell_detail(self, doorbell_id):
-        """Return lock status."""
+        """Return status if the door is locked or unlocked.
+
+        This is status for the lock itself
```
Added period.

ehendrix23 commented on homeassistant/components/august.py:
```diff
@@ -218,17 +221,31 @@ def get_doorbell_detail(self, doorbell_id):
+    def get_lock_door_status(self, lock_id):
+        """Return status if the door is open or closed.
+
+        This is the status from the door sensor
```
Period added.

ehendrix23 commented on homeassistant/components/august.py:
```diff
@@ -237,15 +254,24 @@ def get_lock_detail(self, lock_id):
+        _LOGGER.debug("Start retrieving locks status")
         for lock in self._locks:
-            status_by_id[lock.device_id] = self._api.get_lock_status(
+            _LOGGER.debug("Updating status for %s.",
```
Period removed

ehendrix23 commented on homeassistant/components/lock/august.py:
```diff
@@ -77,6 +81,9 @@ def changed_by(self):
     def device_state_attributes(self):
         """Return the device specific state attributes."""
+        if self._lock_detail is None:
+            return
```
Changed to return None.

balloob committed:
Add a webhook automation trigger (#17246)

damarco committed:
Use only_cache parameter in binary_sensor.zha.Remote (#16711)

JariInc committed:
Add configurable temperature step for MQTT climate component (#16201)

* Add configurable temperature step

* Remove temp step from climate component

balloob committed:
block external IP (#17248)

* block external IP

* Update __init__.py

nhorvath committed:
Thermoworks Smoke Sensor (#16139)

* Add support for monitoring the temperature of a thermoworks smoke thermometer.

* Use string formatting.

* Add line break.

* Add error handling for authentication.

* Fix linting errors.

* Fix quotes.

* Bump thermoworks_smoke library version.

* Review changes for @MartinHjelmare

* Add unique id attribute and change battery attribute to the standard "battery_level".

* requested changes to snake case and monitored conditions

* fix lint error

* exclude firmware from state attrs. rename original_unit to unit_of_min_max so it's more clear what it is for.

* add device_info

* add regex validator for exclude

* undo device info stuff

* remove serial number from attributes even though other components are allowed to have it...

* exclude firmware

rytilahti committed:
Bump python-miio requirement (#17260)

helto4real committed:
Swedish weather institute weather component (#16717)

* SMHI Component

* Clean up typos

* Fixed default values first config to home location (tests will follow)

* Fixed tests and removed unused function

* Minor fixup after comments from @kane610

* add support for precipitation in forecast

* Removed old async_step_init not needed.

amelchio committed:
Remove warning on script delay (#17264)

* Remove warning on script delay

* Use suppress

StevenLooman committed:
Add defaults, fixing #17229 (#17261)

syssi committed:
Fix ambient light state of the Philips Eyecare Lamp (Closes: #16269) (#17259)

syssi committed:
Add basic support of the Philips Zhirui desk lamp (philips.light.mono1) (#17258)

marcolertora committed:
Added lumitek/ankuoo recswitch component (#15764)

* Added lumitek/ankuoo recswitch component

* cosmetics

* remove callback

* cosmetics

* update requirements pyrecswitch==1.0.2

* add in .coveragerc

OttoWinter committed:
Add device registry to MQTT fan (#17250)

OttoWinter committed:
Make async_track_time_change smarter (#17199)

* Make async_track_time_change smarter

* Move to util/dt

* Remove unnecessary check

* Lint

* Remove tzinfo check

* Remove check

* Add comment about async_track_point_in_utc_time

* Fix typing check

* Lint

danielperna84 committed:
Add optional "all" parameter for groups (#17179)

* Added optional mode parameter

* Cleanup

* Using boolean configuration

* Fix invalid syntax

* Added tests for all-parameter

* Grammar

* Lint

* Docstrings

* Better description

definitio committed:
Fix mpd timeout error (#17254)

* Increase mpd client timeout

* Update mpd.py

damarco committed:
Add support for zha custom cluster mappings (#16714)

* Add support for custom cluster mappings

* Refactor sub_component mapping

cgarwood committed:
Z-Wave Config Entry Support (#17119)

* Initial Z-Wave Config Entry Support

* Use conf.get() for config import

* Uncomment test

* Re-add line breaks

* tabs -> space

* Unused import cleanup & lint fixes

* Remove unused config flow link step

* Address comments

* Remove unused import

* Fix tests

* Check for valid usb_path

* Test for Z-Stick in config flow

* Pass config dir to ZWaveOption

* Auto-generate Network Key if none provided

* Test fixes

* Address comments & more start network service registration

* add_executor_job for options.lock()

balloob committed:
Manual updates (#17278)

ehendrix23 committed:
Merge branch 'August-Lock-Door-Binary-Sensor' of https://github.com/ehendrix23/home-assistant into August-Lock-Door-Binary-Sensor

bachya commented:
This PR is borked with commits that have nothing to do with it. Recommend you redo.

ehendrix23 commented:
> This PR is borked with commits that have nothing to do with it. Recommend you redo.

WOW, I can see that. Any idea what I might have done wrong to caused this?

I know I did a rebase of my branch which I thought all came back clean.  Want to ensure that this does not happen again. :-)

armills commented:
There are full instructions here for rebasing: https://developers.home-assistant.io/docs/en/development_catching_up.html Most importantly, after you rebase, you need to force push to your branch on GitHub, and be sure NOT to pull first as git suggests.

You can fix this branch by running `git rebase -i upstream/dev` for interactive mode, then delete any commits that aren't related to your change. Once you're done force push up to GitHub.

ehendrix23 commented:
Going back through my command history, looks like I indeed did a pull as git suggested. :-(

I still have the update files. Might it be better/easier instead to close this PR instead, delete the branch and then create a new branch and a new PR?

armills commented:
Yeah, that works
